lib/cursor.js

```diff
@@ -4,21 +4,35 @@ const common = require('metarhia-common');
 
 const core = require('./core');
 
+const defaultOptions = {
+  jsql: [],
+  parents: [],
+  timeout: 5000,
+  category: null,
+  schema: null,
+  provider: null,
+};
+
 class Cursor {
-  constructor() {
-    this.parents = [];
-    this.provider = null;
+  constructor(options) {
     this.dataset = [];
     this.children = [];
-    this.jsql = [];
-    this.schema = null;
+    Object.keys(defaultOptions).forEach(key => {
+      if (!options || options[key] === undefined) {
+        this[key] = defaultOptions[key];
+      } else {
+        this[key] = options[key];
+      }
+    });
   }
 
   definition(
     // Attach schema
+    category, // string, schema name
```

Comment:
Thus we can't set `schema` without setting `category`. It is a common case when we generate noname schema form one or more named schemas in run-time and will pass to cursor. I'd prefer single argument `.definition(catagoryNameOrSchema)` so if argument is `string` we should find schema in collection `gs.categories.get(categoryName)`

Reply:
Well, PGCursor doesn't make sense without schemaName/Category (It is used as a tableName for now or do I have any other way of determining a tableName?) and if/when we add validation we will need the schema itself.
Also, I don't like the recursive dependency of Cursor on a `gs` and I'd prefer for the user to explicitly and manually get and pass the schema.

Reply:
/cc @tshemsedinov